### PR TITLE
activity: fix triggering pit trap after digging it

### LINKED ISSUE #42049: Fall in pit while digging pit
# Describe the bug

When digging a pit from a shallow pit, you trigger the pit you dug.

# Steps To Reproduce

1. Spawn player and entrenching tool.
2. Dig pit.
3. Dig pit again.
4. Trigger the pit and hurt yourself.

# Expected behavior

You shouldn't trigger the pit since you're digging it.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-3783-g754b89e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Arcana and Magic Items [Arcana],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Mythical Martial Arts [MMA],
    No Rail Stations [No_Rail_Stations],
    Bionic Slots [cbm_slots],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills]
]

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary

`SUMMARY: Bugfixes "Fix falling into a pit right after digging it"`

#### Purpose of change

Digging a deep pit makes you fall in it at the end of the activity

Closes #42049

#### Describe the solution

Move map::may_trigger_trap() from activity::finish to activity::do_turn

#### Describe alternatives you've considered

none

#### Testing

Set PC perception to 0 then dig a few deep pits and confirm PC doesn't trigger the new pit traps
Move to empty tile, use map editor to add a buried bear trap on the same tile as PC, then start digging. Confirm trap still springs

#### Additional context

I'm bad

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -343,13 +343,14 @@ void dig_activity_actor::start( player_activity &act, Character & )
     act.moves_left = moves_total;
 }
 
-void dig_activity_actor::do_turn( player_activity &, Character & )
+void dig_activity_actor::do_turn( player_activity &, Character & who)
```
Comment (kevingranade) [inline src/activity_actor.cpp:346]:
```suggestion
void dig_activity_actor::do_turn( player_activity &, Character &who )
```

File: src/activity_actor.cpp [hunk 2]
```diff
@@ -446,13 +445,14 @@ void dig_channel_activity_actor::start( player_activity &act, Character & )
     act.moves_left = moves_total;
 }
 
-void dig_channel_activity_actor::do_turn( player_activity &, Character & )
+void dig_channel_activity_actor::do_turn( player_activity &, Character & who)
```
Comment (kevingranade) [inline src/activity_actor.cpp:448]:
```suggestion
void dig_channel_activity_actor::do_turn( player_activity &, Character &who )
```

**Comment by andrei8l** [comment]
Sorry about that astyle error

**Comment by wapcaplet** [comment]
Tested OK in your branch. I am able to dig a shallow pit into a deep pit, and do not fall in when it is finished.

image

As expected, if I step back into the pit, I fall in, and have trouble escaping:

image

Looks good to me!

**Pull request merged**